pkg/ceph/client/pool.go

```diff
@@ -124,6 +137,32 @@ func CreatePoolWithProfile(context *clusterd.Context, clusterName string, newPoo
 	return CreatePoolForApp(context, clusterName, newPool, appName)
 }
 
+func DeletePool(context *clusterd.Context, clusterName string, name string) error {
+	// check if the pool exists
+	pool, err := GetPoolDetails(context, clusterName, name)
+	if err != nil {
+		logger.Infof("pool %s not found for deletion. %+v", name, err)
+		return nil
+	}
+
+	logger.Infof("purging pool %s (id=%d)", name, pool.Number)
+	args := []string{"osd", "pool", "delete", name, name, reallyConfirmFlag}
+	_, err = ExecuteCephCommand(context, clusterName, args)
+	if err != nil {
+		return fmt.Errorf("failed to delete pool %s. %+v", name, err)
+	}
+
+	// remove the crush rule for this pool and ignore the error in case the rule is still in use or not found
```

Comment:
should the crush rule be removed before deleting the pool?

Reply:
actually you have to remove the crush rule after the pool or else it is in use